### DISCUSSION TITLE
React Native Firebase is not working at all in release (Android)

### DISCUSSION THREAD

**Discussion by MadCoderme** [invertase/react-native-firebase#4221] [Q&A] [answered]
I am using Firebase Cloud Messaging in my react native project. I have done all the steps mentioned in the DOCS.
Unfortunately, it works perfectly in debug or development mode, but doesn't work at all in release.
I am using the latest version of `@react-native-firebase/messaging` and `React Native`
I already explored Internet, but could not find a solution. Even, I couldn't find a similar issue anywhere.
Please let me know what codes should I share with everyone.
Please help, I need help seriously 

♦The app runs without crash
♦ I couldn't find anything wrong in `adb logcat`, If you want to see it let me know please
♦ Proguard and Hermes both are disabled
♦ Project MINSDK Version 16

**Comment by mikehardy** [comment, marked as answer]
You should probably open an issue and follow the template so we have all the information
For instance, you don't even specifically say it's android, I'm just guessing based on the other info provided.
Very strange it works in dev but not release though, that sounds like a project config issue vs a coding issue

**Reply by MadCoderme** [reply]
Okay, Let me open an issue

**Reply by dariomalfatti-centropaghe** [reply]
Hi, I've the same issue with Android (not yet try with iOS). Did you open the issue where I can read something usefull to me or did you find the solution? Thank you in advance

**Reply by mikehardy** [reply]
@dariomalfatti-centropaghe try --> https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh and/or https://github.com/invertase/react-native-firebase-authentication-example - both are fully automated things that will show you how it's supposed to work. Both work fine for me ios and android, debug and release

**Reply by dariomalfatti-centropaghe** [reply]
Thank you for the fast reply. I'm taking a look later this day to the examples. Anyway I found the solution yesterday. On my RN project I had 2 index files as starting point. One called index.android.js and one index.native.js. My right code was into index.android.js. In debug mode it took the code from index.android but in release mode it took the code from index.native.js. Strange, I thought the index.android would have the priority to all the other files. From the documentation, if I haven't understand wrong, seems that now exists only the index.native, maybe it's a bug that it take the index.android code in debug mode.

**Reply by mikehardy** [reply]
I would guess - not certain but guess - that when the bundler is doing your release build you have it configured with a firm entry point definition to index.native.js so the bundle is only getting that code, but in debug mode metro is able to do platform selection. You might look at the android/app/build.gradle file to see what entry point is configured for release mode

**Reply by dariomalfatti-centropaghe** [reply]
Great to know, that was unknown to me. In fact, I took a look at the build.gradle and my entry point is index.native.js.
Thank you
